app/views/declaration/summary/parties_section.scala.html

```diff
@@ -14,113 +14,124 @@
-                value = representative.statusCode.map(code => messages(s"declaration.summary.parties.representative.type.$code")).getOrElse(""),
-                call = Some(controllers.declaration.routes.RepresentativeDetailsController.displayPage(mode)),
-                changeLabel = Some(messages("declaration.summary.parties.representative.type.change"))
-            ))
-        )
-        @declarationData.parties.carrierDetails.map(carrier =>
-            eori_or_address_section(
-                id = "carrier",
-                eoriOption = carrier.details.eori,
-                eoriLabel = "declaration.summary.parties.carrier.eori",
-                address = carrier.details.address,
-                addressLabel = "declaration.summary.parties.carrier.address",
-                call = Some(controllers.declaration.routes.CarrierDetailsController.displayPage(mode)),
-                eoriChangeLabel = Some(messages("declaration.summary.parties.carrier.eori.change")),
-                addressChangeLabel = Some(messages("declaration.summary.parties.carrier.address.change")),
-                isEoriDefault = false
-            )
-        )
+    @summaryList("declaration-parties-summary", Some(messages("declaration.summary.parties")),
 
-        @declarationData.parties.declarationAdditionalActorsData.map(additional =>
-            parties_section_additional_actors(
-                mode = mode,
-                additionalActors = additional.actors
-            )
-        )
+        declarationData.parties.exporterDetails.map(export => EoriOrAddress.rows(
+            key = "exporter",
+            eori = export.details.eori,
+            address = export.details.address,
+            eoriLabel = "declaration.summary.parties.exporter.eori",
+            eoriChangeLabel = "declaration.summary.parties.exporter.eori.change",
+            addressLabel = "declaration.summary.parties.exporter.address",
+            addressChangeLabel = "declaration.summary.parties.exporter.address.change",
+            changeController = controllers.declaration.routes.ExporterDetailsController.displayPage(mode)
+        )).getOrElse(Seq.empty)
+
+        ++ declarationData.parties.consigneeDetails.map(consignee => EoriOrAddress.rows(
+            key = "consignee",
+            eori = consignee.details.eori,
+            address = consignee.details.address,
+            eoriLabel = "declaration.summary.parties.consignee.eori",
+            eoriChangeLabel = "declaration.summary.parties.consignee.eori.change",
+            addressLabel = "declaration.summary.parties.consignee.address",
+            addressChangeLabel = "declaration.summary.parties.consignee.address.change",
+            changeController = controllers.declaration.routes.ConsigneeDetailsController.displayPage(mode),
+            isEoriDefault = false
+        )).getOrElse(Seq.empty)
+
+        ++ declarationData.parties.declarantDetails.map(declarant => EoriOrAddress.rows(
```

Comment:
I think it can be worth to extract this to some variables to build the whole sequence or extract every element to the variable and just add it here. Inlined version is difficult to read

Reply:
Yes.  The bulk of the parameters to the EoriOrAddress.rows() helper are the same.   I'll re-factor